actors.md

```diff
@@ -1091,6 +1092,46 @@ func ChangeWorker(addr Address) {
 }
 ```
 
+#### `PaymentVerify`
+
+Verifies a storage market payment channel voucher's 'Extra' data.
+
+**Parameters**
+
+```sh
+type PaymentVerify struct {
+    Extra StorageVoucherData
+    Proof Bytes
+} representation tuple
+
+type StorageVoucherData struct {
+    DataCommitment Bytes
+    RequiredSector Bytes
```

Comment:
Theres many reasons we can end up with this. The most straightforward is when a client wants to extend their deal with a miner. They just create new payment channel vouchers that reference the sector directly.

In other cases, with payment reconciliation, a miner may go back to a client and ask for a new set of vouchers, offering some small discount in exchange for vouchers that reference the sector directly (so what ends up happening is the client and miner can split the difference of the fees that would have been paid for submitting all that extra data to the chain)

Reply:
These scenarios where valid vouchers are created with a sector id make sense, but the base case (AFAIK) is that the miner must have valid vouchers for the duration of a storage deal prior to accepting the deal. This means prior to the creation of the sector and the sector id. In this case wouldn't the miner still need a way to provide a sector id in order to collect payment (or ward off a premature channel closure by a bad actor client)?

I don't see how the client provided sector id scenarios provide enough protection for the miner.

Reply:
@acruikshank oh no, at first, the payments only reference the piece commitment, and the miner must pass a PIP to prove they have the data to cash out. No sector ID involved.

Look in the `PaymentVerify` method, if a DataCommitment is present, we check the proof against the data commitment. Otherwise, if a RequiredSector is present, we verify that exists.

Reply:
If the miner is in possession of vouchers without a RequiredSector and the client's piece, can't they simply create a fake sector containing the piece, generate a PIP from it, throw away the sector, error out on the deal, and keep the proof? The client wouldn't be able to cancel the the channel because the miner would have valid vouchers. The client also wouldn't be able to arbitrate the deal, because the miner never completed it.

Reply:
the vouchers are only valid if a *live* sector matches the generated PIP. The client doesnt really care what sector the data is in, as long as its stored.

The scenario you mention here implies that the miner still has the clients

Reply:
> if a live sector matches the generated PIP.
 
It sounds like you're expecting the sector to be derivable from the proof. If the proof contained CommD (the current rust-fil-proofs construction doesn't) we could reverse search the sector commitments for the sector id to prove liveness. 

Still seems more efficient for the miner to send the sector id instead of CommD. 

I'm also still confused about the point of the `RequiredSector` if we're assuming the sector is derivable from the proof. Would a client want to pin a piece to a sector?

Reply:
Oh, I see now you are expecting the pip to have a sector ID, so 👍